proposed/2023/Add-dotnet-search.md

```diff
@@ -41,23 +44,91 @@ This option will specify a list of sources to search from. If a source is not sp
-|---------------------------------------------|----------------|---------|-----------------|
-| Newtonsoft.Json                             | 13.0.3         |         | 3,829,822,911   |
+        Source: nuget.org
+        | Package ID                                  | Latest Version | Authors | Downloads       |
+        |---------------------------------------------|----------------|---------|-----------------|
+        | Newtonsoft.Json                             | 13.0.3         |         | 3,829,822,911   |
+
+#### **Option `--format`**
+
+This option will allow the specification of the output format. The option will take one of these three arguments: Table, List, or Json. The outputs will depend on verbosity. The following examples are for normal verbosity.
+
+- `--format Table` the output will be in a tabular form as shown below
+
+        Source: nuget.org
+        | Package ID                                  | Latest Version | Authors | Downloads       |
+        |---------------------------------------------|----------------|---------|-----------------|
+        | Newtonsoft.Json                             | 13.0.3         |         | 3,829,822,911   |
+- `--format List` : each package will be listed line by line
+
+        >Newtonsoft.Json | 13.0.3 | Authors: N/A | Downloads: 3,829,822,911
+        >Microsoft.IdentityModel.JsonWebTokens | 7.0.3 Authors: N/A | Downloads: N/A
+
+- `--format Json` : A json output will be outputted. It will contain the following metadata.
+  - Source
+  - Package ID
+  - LatestVersion
+  - Authors
+  - Downloads
+
+  The output will take the following format
+
+        [
+            {
+                "source": "source Name1",
+                "packagesList": [
+                    {
+                        "author": "",
+                        "downloads": "",
+                        "latestVersion": "",
+                        "packageId": ""
+                    }
+                ]
+            },
+            {
+                "source": "source Name2",...
+            }
+        ]
+
+#### Option `--verbosity`
+
+  controls the amount of information included in the output of a command. This allows you to tailor the output to your needs, providing either a concise summary or a detailed breakdown of the information returned by the command. 
+
+Here are the details of each verbosity level:
+
+- **`--verbosity quiet`** : The output will contain minimal information. For each package, it will display only the following metadata:
+  - Package ID
+  - Latest Version
+  
+- **`--verbosity normal`** : The output will provide a moderate amount of information. For each package, it will display the following metadata:
+  - Package ID
+  - Latest Version
+  - Authors
+  - Downloads
+  
+- **`--verbosity detailed`** : The output will be highly detailed. For each package, it will display the following metadata:
+  - Package ID
+  - Latest Version
+  - Authors
```

Comment:
This was a miss in the original write-up, but we do have a discussion of owners vs authors that should probably factor in here. 

cc @albarry4 @donnie-msft

Reply:
@joelverhagen what's are your thoughts? 

fyi @aortiz-msft @mariaghiondea

Reply:
IMO owners should be shown if available. It is a credible value compared to authors and aligns with current PM UI work. CC @donnie-msft 

Client should detect the presence of `owners` property in search results. Note that packages can have zero owners so you need to detect presence of the property not whether the owners array is empty.
Problem is registration API does not contain owners. It's only on the search API. Meaning `--exact-match`

There is inconsistency in .NET CLI already.
`dotnet new search cake` -> owners
`dotnet tool search cake` -> authors

Reply:
In PM UI, the approach I'm proposing is `<package id> "by" <owner if available, or author if not>`.
If the CLI output (eg, for table format) had a column header **"By"**, and it did the same logic, that would make for consistency between CLI and VS.

Reply:
In PM UI, something will indicate that it's connected to a nuget.org owner. Not sure what that might be in CLI output. 

#### Author idea for CLI:
_source: private feed (Author)_
|Package ID|By|
|--|--|
|Newtonsoft.Json|James Newton-King|

Customers need to understand that absense of an indicator means it's not an "owner". Could add a footnote to explain?

#### Owner idea 1 for CLI:
_source: nuget.org (Owner)
|Package ID|By|
|--|--|
|Newtonsoft.Json|dotnetfoundation (nuget.org), jamesnk (nuget.org), newtonsoft (nuget.org)|
...

#### Owner idea 2 for CLI:
_source: nuget.org (Owner)
|Package ID|By|
|--|--|
|Newtonsoft.Json|dotnetfoundation*, jamesnk*, newtonsoft*|
...
*Owner according to package source

Reply:
Updated the spec. After yesterday's discussion, it was not favored to show owners and Authors both together. However, it was clear that we want to show `Owners` property. As a result, we will be showing `Owners` in this iteration, and we can decide on whether and how we can include authors information later if there is a request for it.